sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt

```diff
@@ -0,0 +1,386 @@
+                searchQueryValueChange = {
+                    metadata.value =
+                        metadata.value.copy(searchQuery = it)
+                },
+                onCloseSearchFieldClick = {
+                    metadata.value =
+                        metadata.value.copy(isSearchActive = false)
+                },
+                onClearSearchField = {
+                    metadata.value =
+                        metadata.value.copy(searchQuery = "")
+                },
+            )
+        },
+        actions = {
+            ShowkaseAppBarActions(
+                metadata,
+                currentRoute,
+            )
+        },
+    )
+}
+
+@Suppress("LongParameterList")
+@Composable
+private fun SampleAppBarTitle(
+    isSearchActive: Boolean,
+    group: String?,
+    componentName: String?,
+    componentStyleName: String?,
+    route: String?,
+    searchQuery: String?,
+    searchQueryValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    onCloseSearchFieldClick: () -> Unit,
+    onClearSearchField: () -> Unit,
+) {
+
+    AnimatedVisibility(
+        visible = isSearchActive,
+        enter = expandHorizontally(),
+        exit = shrinkHorizontally(),
+    ) {
+        SearchField(
+            searchQuery = searchQuery,
+            searchQueryValueChange = searchQueryValueChange,
+            onCloseSearchFieldClick = onCloseSearchFieldClick,
+            onClearSearchField = onClearSearchField,
+        )
+    }
+    AnimatedVisibility(
+        visible = !isSearchActive,
+        enter = slideInHorizontally() + expandIn(),
+    ) {
+        AppBarTitle(
+            currentRoute = route,
+            modifier = modifier,
+            currentGroup = group,
+            currentComponentName = componentName,
+            currentComponentStyleName = componentStyleName,
+        )
+    }
+}
+
+@Composable
+private fun AppBarTitle(
+    modifier: Modifier,
+    currentRoute: String?,
+    currentGroup: String?,
+    currentComponentName: String?,
+    currentComponentStyleName: String?,
+) {
+    when {
+
+        currentRoute == CurrentScreen.COMPONENT_GROUPS.name -> {
+            ToolbarTitle(stringResource(R.string.components_category), modifier)
+        }
+
+        currentRoute.insideGroup() -> {
+            ToolbarTitle(currentGroup ?: "currentGroup", modifier)
+        }
+
+        currentRoute == CurrentScreen.COMPONENT_STYLES.name -> {
+            ToolbarTitle(currentComponentName.orEmpty(), modifier)
+        }
+
+        currentRoute == CurrentScreen.COMPONENT_DETAIL.name -> {
+            val styleName = currentComponentStyleName?.let { "[$it]" }.orEmpty()
+            ToolbarTitle(
+                string = "${currentComponentName.orEmpty()} $styleName",
+                modifier = modifier,
+            )
+        }
+    }
+}
+
+
+@Composable
+public fun ToolbarTitle(
+    string: String,
+    modifier: Modifier,
+) {
+    val lineCount = remember { mutableStateOf(0) }
```

Comment:
Why are we interested in line's count?

Reply:
I don't remember precisely but I think it's only used for semantics